source/core/constraint/unique.d

```diff
@@ -13,45 +13,177 @@ import core.sudoku.cell;
  */
 public class UniqueConstraint : Constraint
 {
-	private this(Cell[] cells...)
+	private this(Cell cell)
 	{
 		super (ConstraintType.Unique);
-		this.cells ~= cells;
+		this.cells ~= cell;
 	}
 
 
-	public static auto createSingle(Cell cell1, Cell cell2)
+	/** Connects two Cells
+	 *
+	 * Ensures `cell1` is connected to `cell2` and vice-versa
+	 * This means when a digit is checked for it's availability in `cell1`
+	 *     the validation is dependent of the digit in `cell2` and vice-versa
+	 * A Cell cannot connect to itself
+	 *
+	 * Params:
+	 *     cell1 = cell to connect
+	 *     cell2 = cell to connect
+	 *
+	 * Examples:
+	 * --------------------
+	 * Cell cell1 = new Cell(0);
+	 * Cell cell2 = new Cell(0);
+	 * UniqueConstraint.createSingle(cell1,cell2);
+	 * assert(cell1.get!UniqueConstraint.cells.equal([cell2]));
+	 * assert(cell2.get!UniqueConstraint.cells.equal([cell1]));
+	 * --------------------
+	 */
+	public static void createSingle(Cell cell1, Cell cell2)
```

Comment:
Are you ensuring in the create methods (`createSingle`, `createMultiSingle`, `createInterconnected`) that a cell cannot connect to itself? *(since you're checking that explicitely in the remove methods, but not in these)*

Reply:
I check that in the connect function from Cell, `if (cell != this)`. But saying that, yes I forgot to add it here aswell, although it works, it might have an unecessary function call if the Cells are the same.